addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoBrainHandler.java

```diff
@@ -0,0 +1,389 @@
+
+            NeeoUtil.checkInterrupt();
+            neeoBrainApi = api;
+
+            final Map<String, String> properties = new HashMap<>();
+            addProperty(properties, "Name", brain.getName());
+            addProperty(properties, "Version", brain.getVersion());
+            addProperty(properties, "Label", brain.getLabel());
+            addProperty(properties, "Is Configured", String.valueOf(brain.isConfigured()));
+            addProperty(properties, "Key", brain.getKey());
+            addProperty(properties, "AirKey", brain.getAirkey());
+            addProperty(properties, "Last Change", String.valueOf(brain.getLastChange()));
+            updateProperties(properties);
+
+            if (config.isEnableForwardActions()) {
+                NeeoUtil.checkInterrupt();
+
+                forwardActionServlet = new NeeoForwardActionsServlet(new NeeoForwardActionsServlet.Callback() {
+                    @Override
+                    public void post(String json) {
+                        triggerChannel(NeeoConstants.CHANNEL_BRAIN_FOWARDACTIONS, json);
+
+                        final NeeoAction action = gson.fromJson(json, NeeoAction.class);
+
+                        for (final Thing child : getThing().getThings()) {
+                            final ThingHandler th = child.getHandler();
+                            if (th instanceof NeeoRoomHandler) {
+                                ((NeeoRoomHandler) th).processAction(action);
+                            }
+                        }
+                    }
+
+                }, config.getForwardChain());
+
+                NeeoUtil.checkInterrupt();
+                try {
+                    servletPath = NeeoConstants.WEBAPP_FORWARDACTIONS.replace("{brainid}", brainId);
+
+                    httpService.registerServlet(servletPath, forwardActionServlet, new Hashtable<>(),
+                            httpService.createDefaultHttpContext());
+
+                    final URL callbackURL = createCallbackUrl(brainId, config);
+                    if (callbackURL == null) {
+                        logger.debug(
+                                "Unable to create a callback URL because there is no primary address specified (please set the primary address in the configuration)");
+                    } else {
+                        final URL url = new URL(callbackURL, servletPath);
+                        api.registerForwardActions(url);
+                    }
+                } catch (NamespaceException | ServletException e) {
+                    logger.debug("Error registering forward actions to {}: {}", servletPath, e.getMessage(), e);
+                }
+            }
+
+            NeeoUtil.checkInterrupt();
+            updateStatus(ThingStatus.ONLINE);
+            NeeoUtil.checkInterrupt();
+            if (config.getCheckStatusInterval() > 0) {
+                NeeoUtil.cancel(checkStatus.getAndSet(scheduler.scheduleWithFixedDelay(() -> {
```

Comment:
Why do you cancel this task immediately?

Reply:
I cancel the prior task if it's been scheduled (I get the existing task, then set the new task [atomically with the get] and then cancel the existing task if there was one).